server/devdocs/selectors/index.js

```diff
@@ -23,47 +29,105 @@ const SELECTORS_DIR = path.resolve( __dirname, '../../../client/state/selectors'
 const router = express.Router();
 let prepareFuse;
 
-function parseSelectorFile( file ) {
-	return new Promise( ( resolve, reject ) => {
-		fs.readFile( SELECTORS_DIR + '/' + file, 'utf8', ( error, contents ) => {
-			if ( error ) {
-				return reject( error );
+const jsParse = contents => {
+	try {
+		return espree.parse( contents, {
+			attachComment: true,
+			ecmaVersion: 6,
+			sourceType: 'module',
+			ecmaFeatures: {
+				experimentalObjectRestSpread: true,
 			}
+		} );
+	} catch ( e ) {
+		console.log( e.message );
+		return null;
+	}
+};
+
+const parseSelectorFile = file => new Promise( resolve => {
+	const contents = fs.readFileSync( path.resolve( SELECTORS_DIR, file ), { encoding: 'utf8' } );
+	const ast = jsParse( contents );
+
+	if ( ! ast ) {
+		console.warn(
+			chalk.red( '\nWarning: ' ) +
+			chalk.yellow( 'Could not parse file: ' ) +
+			chalk.blue( path.basename( file ) )
+		);
+		return resolve( null );
+	}
 
-			const selector = {
-				name: camelCase( path.basename( file, '.js' ) )
-			};
+	const expectedName = camelCase( path.basename( file, '.js' ) );
+
+	// if the default export has a comment attached to it
+	// then we don't need to jump to the referenced declaration
+	const moduleExport = defaultExport( ast );
+	const defaultComment = getComment( moduleExport );
+	const defaultNode = defaultComment
+		? moduleExport
+		: defaultFunction( ast );
+
+	if ( ! defaultNode ) {
+		console.warn(
+			chalk.red( '\nWarning: ' ) +
+			chalk.yellow( 'Could not find default-exported function' ) +
+			chalk.yellow( '\nBased on the filename: ' ) +
+			chalk.blue( path.basename( file ) ) +
+			chalk.yellow( '\nWe expected to find an exported function with name ' ) +
+			chalk.blue( expectedName )
+		);
+		return resolve( null );
+	}
 
-			forEach( contents.match( REGEXP_DOCBLOCKS ), ( docblock ) => {
-				const doc = doctrine.parse( docblock, { unwrap: true } );
-				if ( doc.tags.length > 0 ) {
-					Object.assign( selector, doc );
-					return false;
-				}
-			} );
+	const comment = getComment( defaultNode );
+	if ( ! comment ) {
+		console.warn(
```

Comment:
Well the way I saw it is that if those warnings trigger then we have selectors which can't be found, meaning that we should complain loudly. If we disable the warnings then I'm sure that will spell the end of the hope of fixing them.

_maybe_ we could grab the `git` blame for the file or lines…